command_definitions.py

```diff
@@ -0,0 +1,128 @@
+import glob
+import re
+
+from fabric.api import run, put
+
+from host_setup import DOCKER_COMPOSE_EXECUTABLE, mkdir, provide_docker_compose
+
+
+class CallList(object):
+    def __init__(self, command, predicate=lambda: True):
+        self.command_list = command
+        self.base_command = ['echo']
+        self.predicate = predicate
+
+    def execute(self, project, hostname, environment=None):
+        if environment is None:
+            environment = {}
+        self.base_command = [
+            '{}={}'.format(env_var_name, env_var_value) for env_var_name, env_var_value in environment.items()
+        ] + self.base_command
```

Comment:
Probably shouldn't modify `self.base_command`. A `Command` object should support being `execute()`d several times without changing what command it runs.